src/servers/grpc_server_v2.cc

```diff
@@ -514,6 +515,394 @@ Handler<ServiceType, ServerResponderType, RequestType, ResponseType>::Stop()
   LOG_VERBOSE(1) << "Thread exited for " << Name();
 }
 
+template <typename ResponderType, typename RequestType, typename ResponseType>
```

Comment:
This seems like much more code than the existing solution. It does use a single thread but it seems like there is likely something less complex? No need to change now but we should revisit when we put all the other low-overhead APIs onto this single thread.